### PR TITLE
feat: scrape Stop&Shop

### PR DESCRIPTION
**Pull request by tkjones117**
In the same way we scrape Hannaford (just check if there's availability), add scraper for Stop & Shop.

These locations come from vaxfinder: https://vaxfinder.mass.gov/?zip_or_city=&q=stop

### CODE REVIEW

**Comment by tkjones117** [comment]
NOTE: Putting as draft b/c I'm getting the following error when I try to run `node main.js StopAndShop`:

`Error: Protocol error (Network.getResponseBody): No resource with given identifier found`

File: site-scrapers/Hannaford.js [hunk 1]
```diff
@@ -7,7 +7,6 @@ module.exports = async function GetAvailableAppointments(browser) {
     const webData = await ScrapeWebsiteData(browser);
     console.log("Hannaford done.");
     return sites.Hannaford.locations.map((loc) => {
-        const newLoc = { ...loc };
```
Comment (tkjones117) [inline site-scrapers/Hannaford.js:10]: This was an unused var.

File: site-scrapers/StopAndShop.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        if (!results[loc.zip]) {
+            await page.evaluate(
+                () => (document.getElementById("zip-input").value = "")
+            );
+            await page.type("#zip-input", loc.zip);
+            const [searchResponse, ...rest] = await Promise.all([
+                Promise.race([
+                    page.waitForResponse(
+                        // TODO - this link doesn't work for me... fix? 
+                        "https://stopandshopsched.rxtouch.com/rbssched/program/covid19/Patient/CheckZipCode"
```
Comment (tkjones117) [inline site-scrapers/StopAndShop.js:46]: This site doesn't work after the first time I visit it in my browser... wonder if that's related to my issue. But I just followed what we do for Hannaford, which doesn't generate the error.

File: site-scrapers/index.js [hunk 3]
```diff
@@ -14,7 +14,7 @@ ls.map((fileName) => {
 });
 
 if (process.argv.length > 2) {
-    scrapers = []; // args override directory list allowing single site runs
+    scrapers = []; // args override directory list allowing single site runs, e.g. `node main.js LynnTech`
```
Comment (tkjones117) [inline site-scrapers/index.js:17]: adding for clarity - unrelated to the PR's main purpose

File: site-scrapers/StopAndShop.js (hunk 2 above)
Comment (tkjones117) [reply site-scrapers/StopAndShop.js:48]: This is the underlying API call made from the site - so it doesn't matter if it doesn't work when I visit the url directly in my browser!

**Comment by tkjones117** [comment]
Fixed the issue. Turns out the site was rate limiting me after 10 requests. 
Running chromium with headless=false showed me the error, that the user had made too many requests. I continued to get this error when I visited in my browser, until I switched to incognito, which suggested this was a cookie issue. After deleting the relevant cookie (`await page.deleteCookie({ name: "ASP.NET_SessionId" });`) I no longer get rate limited!

**Review by johnhawkinson — APPROVED** [review]
Looks ready to deploy.

I continue to have reservations about the weird errors without the cookie deletion and the common code with `Hannaford.js` to refactor, but those are not blockers.

**Comment by tkjones117** [comment]
Added try/catch message to facilitate debugging if we get a protocol error again. 

We can create common code with Hannaford if we have a 3rd location that relies on the same code. I think it's unneeded until then.

@livgust this is rebased and ready for review.

File: data/sites.json [hunk 4]
```diff
@@ -49,6 +49,77 @@
...
+                "city": "Feeding Hills",
+                "zip": "01030"
+            },
+            {
+                "street": "341 Plymouth Road",
+                "city": "Halifax",
+                "zip": "02338"
+            },
+            {
+                "street": "111 Chatam Rd",
```
Comment (livgust) [inline data/sites.json:77]: Chatham

File: site-scrapers/StopAndShop.js [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+                    page.waitForResponse(
+                        "https://stopandshopsched.rxtouch.com/rbssched/program/covid19/Patient/CheckZipCode"
+                    ),
+                    page.waitForNavigation(),
+                ]),
+                page.click("#btnGo"),
+            ]);
+            try {
+                const result = (await searchResponse.buffer()).toString();
+                //if there's something available, log it with a unique name so we can check it out l8r g8r
```
Comment (livgust) [inline site-scrapers/StopAndShop.js:56]: So I just added a folder to our s3 bucket called "debug." I suggest we publish the screenshot there (or locally if we're in development mode). If it's easy, could you update for Hannaford too?

File: site-scrapers/StopAndShop.js [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+                    const filename =
+                        "stopandshop-zip-" +
+                        loc.zip +
+                        "-date-" +
+                        today +
+                        ".png";
+                    await page.screenshot({ path: filename });
+                }
+                results[loc.zip] = result;
+                console.log(`Done with ${loc.zip} ${loc.city}`);
```
Comment (livgust) [inline site-scrapers/StopAndShop.js:74]: Mind removing this extra console statement?

**Review by livgust — CHANGES_REQUESTED** [review]
Good find re: the 10-query limit! I ran this locally with no issue. Couple small comments and suggested change to saving the screenshot (since that won't save in Lambda - things are trashed when they're done IIRC)

**Comment by livgust** [comment]
Hey @tkjones117 , do you want me to plan on merging this in now-ish or should I wait until @zpeyton 's changes are final for Hannaford (so you both can do the S3 upload)?

**Comment by tkjones117** [comment]
@livgust that sounds great, yes let's merge this as-is and I can add the S3 upload later after his changes are in.

**Comment by livgust** [comment]
@tkjones117 you should be all set - s3 upload is confirmed to be working!

File: site-scrapers/StopAndShop.js (hunk 5 above)
Comment (tkjones117) [reply site-scrapers/StopAndShop.js:59]: Hannaford was done in a diff PR. Added S3 upload here.

**Comment by tkjones117** [comment]
@livgust I can't get my permissions to work just yet, so I can't test the S3 upload. But it's doing the same thing as Hannaford and you can try it out if you want? Just want to get this up and working for now.

Next steps:
- merge Hannaford and Stop&Shop code paths
- fix S3 upload - I don't think it actually works when we see availability in prod (I saw avail this morning, and nothing is in the prod bucket `debug` folder)
- start parsing actual data from Hannaford and S&S

**Comment by tkjones117** [comment]
Diagnosed the s3 issue and successfully tested this.

**Comment by tkjones117** [comment]
@livgust Steps 1, 2, and 3 are done and this is ready for review. @zpeyton it might make sense for you to review this when you have time.

**Comment by tkjones117** [comment]
To be safe, I defaulted hasAvailability to false because until we address https://github.com/livgust/covid-vaccine-scrapers/issues/84 we won't be able to differentiate a Protocol error due to rate limiting (unlikely with cookie deletion) vs a Protocol error due to navigating to an appointment selection page.

**Review by livgust — APPROVED** [review]

**Pull request merged**